### DISCUSSION TITLE
Access Tailwind Properties Through CSS Variables

### DISCUSSION THREAD

**Discussion by akashshyamdev** [tailwindlabs/tailwindcss#5746] [Help] [answered]
I'd like to access the value of `h-20`(the class), i.e if:

```
// can be any value for h-20, can I access this through css variables? something like var(tw-h-20)?
height: 2rem;
```

I would like to use this inside a calc function like so:

```
height: calc(100vh - var(tw-h-20));
```

More specifically, I'm building a navbar and main inside a body tag. I want the main to take up the remaining page width after the nav. I don't want to specify a height for the nav. Is there a way around this using tailwind and/or vanilla CSS?

**Comment by dance2die** [comment, marked as answer]
You can use `theme()` function to access TW classes via "dot" notation.  
https://tailwindcss.com/docs/functions-and-directives#theme

Instead of `height: calc(100vh - var(tw-h-20));`,
you can do `height: calc(100vh - theme(spacing.20));`

Here is a  demo https://play.tailwindcss.com/1c7D2EaWrd?layout=horizontal&file=css

**Reply by akashshyamdev** [reply]
Ok that worked 👍 

Also, can you help me with this?

>More specifically, I'm building a navbar and main inside a body tag. I want the main to take up the remaining page width after the nav. I don't want to specify a height for the nav. Is there a way around this using tailwind and/or vanilla CSS?

**Reply by dance2die** [reply]
> I want the main to take up the remaining **page** width after the nav. I don't want to specify a **height** for the nav.

Lost here as you want to adjust `main` **width** while the `nav`'s **height** changes.  

Do you have any sketches/illustrations/figma etc to demo how they need to adjust? Also how they should be responsive according to breakpoints?  For rough sketches you can also use https://excalidraw.com/

**Reply by akashshyamdev** [reply]
![Screenshot 2021-10-12 at 10 00 13](https://user-images.githubusercontent.com/56759828/136891695-bc52d436-f613-4145-97ca-30781fe692a3.png)

@dance2die Currently the `main` is talking the height of the content inside it, whereas I need it to cover the who page below the nav.

**Reply by dance2die** [reply]
depending on structure and how the whole site is designed it can be implemented differently.  

In this following [updated demo](https://play.tailwindcss.com/Mmk9FUjAyL), I wrapped `nav` and `main` with `flex, flex-col`.
The height of nav is decided by padding only (removed `.nav` and inlined)

Importantly, `main`'s width is decided with `w-full` and the full height is achieved with `flex-1` (because the parent container is a flex container)

**Reply by akashshyamdev** [reply]
Thanks 👍
